dali/python/backend_impl.cc

```diff
@@ -2004,7 +2006,12 @@ PYBIND11_MODULE(backend_impl, m) {
           DALI_ENFORCE(meta,
               "Operator " + op_name + "  not found or does not expose valid metadata.");
           return ReaderMetaToDict(meta);
-        });
+        })
+    // On the pipeline destruction we need to release the GIL to shutdown the executor
+    // This way, Python operators that might be still running do not deadlock
+    .def("Shutdown", [](Pipeline *p) {
```

Comment:
I wonder if `Shutdown` shouldn't be an internal method?

Reply:
I think we can do it a bit more automatically using [this](https://github.com/pybind/pybind11/issues/1446) approach.
The `init` of the `Pipeline` Python class returns a smart pointer. We can add a custom deleter to it that would release the lock for the time of `p->Shutdown();` and then delete the pipeline. I think it is way less error-prone than exposing part of the teardown sequence to Python.

Reply:
it should be, although i want to call it from the `__del__` method of Python pipeline object. I could rename it to _Shutdown, though. 

And it is not exposed through the Pyton pipeline wrapper

Reply:
Ok, that's sounds like a good approach